From ef6de6ad85f24e112a6460cdefcfa0ea831c2b17 Mon Sep 17 00:00:00 2001
From: Hui Zhou <hzhou321@anl.gov>
Date: Thu, 1 May 2025 11:47:41 -0500
Subject: [PATCH] hydra: fix singleton init

The previous fix 7e87562555 broke the singleton init. Singleton init is
allowed to proceed without executables in the command line.
---
 src/pm/hydra/mpiexec/mpiexec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pm/hydra/mpiexec/mpiexec.c b/src/pm/hydra/mpiexec/mpiexec.c
index 645478616be..bd38d50a30e 100644
--- a/src/pm/hydra/mpiexec/mpiexec.c
+++ b/src/pm/hydra/mpiexec/mpiexec.c
@@ -56,7 +56,7 @@ int main(int argc, char **argv)
         }
     }
 
-    if (!HYD_uii_mpx_exec_list) {
+    if (!HYD_uii_mpx_exec_list && !HYD_server_info.is_singleton) {
         HYDU_ERR_SETANDJUMP(status, HYD_INVALID_PARAM,
                             "No executable provided. Try -h for usages.\n");
     }
